Zotero/Scenes/Detail/PDF/Views/PDFAnnotationsViewController.swift

```diff
@@ -164,113 +183,94 @@ final class PDFAnnotationsViewController: UIViewController {
             guard let self else { return }
 
             if let keys = state.loadedPreviewImageAnnotationKeys {
-                updatePreviewsIfVisible(for: keys)
+                updatePreviewsIfVisible(state: state, tableView: tableView, for: keys)
             }
 
+            let isVisible = parentDelegate?.isSidebarVisible ?? false
             if let key = state.focusSidebarKey, let indexPath = dataSource.indexPath(for: key) {
-                let isVisible = parentDelegate?.isSidebarVisible ?? false
                 tableView.selectRow(at: indexPath, animated: isVisible, scrollPosition: .middle)
             }
-
+            if state.changes.contains(.sidebarEditing) {
```

Comment:
Delete this. It was applied before `dataSource.apply(snapshot)` previously and it animated nicely together with the data source change. In comment below I describe how it could look in `reloadIfNeeded`. If this doesn't animate properly anymore, then yes, keep it here and remove it there (you still have one instance there).

Reply:
Check https://github.com/zotero/zotero-ios/pull/1040/commits/19acd1e7612bb3e5dcc261eeb6f3c6f0b963d740
Editing change animation seems as smooth as current production version now.